src/aktualizr_repo/repo_test.cc

```diff
@@ -98,6 +98,48 @@ TEST(aktualizr_repo, copy_image) {
   EXPECT_EQ(director_targets["signed"]["targets"].size(), 1);
 }
 
+/*
+ * Add simple delegation
+ * Add image with delegation
+ */
+TEST(aktualizr_repo, delegation) {
+  TemporaryDirectory temp_dir;
+  std::ostringstream keytype_stream;
+  keytype_stream << key_type;
+  std::string cmd = generate_repo_exec + " generate " + temp_dir.Path().string() + " --keytype " + keytype_stream.str();
+  std::string output;
+  int retval = Utils::shell(cmd, &output);
+  if (retval) {
+    FAIL() << "'" << cmd << "' exited with error code\n";
+  }
+  cmd = generate_repo_exec + " adddelegation " + temp_dir.Path().string() + " --keytype " + keytype_stream.str();
+  cmd += " --dname test_delegate --dpattern 'tests/*.txt' ";
+  retval = Utils::shell(cmd, &output);
+  if (retval) {
+    FAIL() << "'" << cmd << "' exited with error code\n";
+  }
+
+  EXPECT_TRUE(boost::filesystem::exists(temp_dir.Path() / "repo/image/test_delegate.json"));
+  auto targets = Utils::parseJSONFile(temp_dir.Path() / "repo/image/targets.json");
+  EXPECT_EQ(targets["signed"]["delegations"]["roles"][0]["name"].asString(), "test_delegate");
+  EXPECT_EQ(targets["signed"]["delegations"]["roles"][0]["paths"][0].asString(), "tests/*.txt");
+
+  cmd = generate_repo_exec + " image " + temp_dir.Path().string() + " --keytype " + keytype_stream.str();
+  cmd += " --dname test_delegate --filename tests/CMakeLists.txt";
+  retval = Utils::shell(cmd, &output);
+  if (retval) {
+    FAIL() << "'" << cmd << "' exited with error code\n";
+  }
+  Uptane::Root root(Uptane::RepositoryType::Image(), Utils::parseJSONFile(temp_dir.Path() / "repo/image/root.json"));
+  auto test_delegate = Utils::parseJSONFile(temp_dir.Path() / "repo/image/test_delegate.json");
+  Uptane::Targets delegate_targets(Uptane::RepositoryType::Image(), test_delegate, root);
+  EXPECT_EQ(delegate_targets.targets.size(), 1);
+  EXPECT_EQ(delegate_targets.targets[0].filename(), "tests/CMakeLists.txt");
+  EXPECT_EQ(delegate_targets.targets[0].length(), 8370);
+  EXPECT_EQ(delegate_targets.targets[0].sha256Hash(),
+            "9cb7c207dfda2cbdc2378c32ee4f77260fadd5f3f2671af2985074f66cde0b5a");
+}
```

Comment:
While I applaud using what is already available, we will have to edit this test every time a change is made to `tests/CMakeLists.txt`. Perhaps it would be better to reuse a different test data file we already have in the repo?